akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorContextAdapter.scala

```diff
@@ -102,8 +103,13 @@ import akka.actor.typed.Behavior.UntypedBehavior
         // TODO dispatcher from props
         ActorRefAdapter(ctx.actorOf(b.untypedProps))
       case _ ⇒
-        Behavior.validateAsInitial(behavior)
-        ActorRefAdapter(ctx.actorOf(PropsAdapter(() ⇒ behavior, props)))
+        try {
+          Behavior.validateAsInitial(behavior)
+          ActorRefAdapter(ctx.actorOf(PropsAdapter(() ⇒ behavior, props)))
+        } catch {
+          case ex: ConfigurationException if ex.getMessage.startsWith("configuration requested remote deployment") ⇒
+            throw new ConfigurationException("Remote deployment not allowed for typed actors", ex)
```

Comment:
not so nice to rely on the exception messages, but there is a test that will fail if we change